### ISSUE TITLE
Patch DXVA AV1 support into mingw headers

### ISSUE THREAD

**Issue by rmkimathi** [m-ab-s/media-autobuild_suite#1987]
How can I apply dxva av1 hardware patch to ".\msys64\mingw64\x86_64-w64-mingw32\include\dxva.h" ... or is possible to fix m-ab-s with this?

As done by BtbN/FFmpeg-Builds here:

https://github.com/BtbN/FFmpeg-Builds/commit/0df2644223a5d3f775b4e25d027a3dc4446caefe

**Comment by 1480c1** [comment]
@biswa96, do you know if we can port that commit to the current mingw-w64 headers package?

**Comment by rmkimathi** [comment]
Thanks. I am just an advanced user trying to test out intel Xe graphics features. I ran ffmpeg -h decoder=av1 and realized dxva is not working. I would really appreciate help from the developers.

**Comment by Biswa96** [comment]
> do you know if we can port that commit to the current mingw-w64 headers package?

tl;dr No, can not.

That patch was proposed in wine mailing list. It contains definitions from Windows Insider SDK which is unstable. So, wine can not apply that patch. Whereas mingw-w64 imports that header from wine and also can not/should not apply things from unstable Insider Build.

If we need that patch we can apply it in MINGW-packages, hope it does not break other things.

**Comment by Biswa96** [comment]
BTW, which project requires this change in header file?

**Comment by 1480c1** [comment]
It seems FFmpeg gates av1 hwaccel for dxva by looking for `DXVA_PicParams_AV1`, which is provided in the patch

**Comment by Biswa96** [comment]
Would you like to do the honor of sending the patch file to msys2/mingw-packages repository?

**Comment by 1480c1** [comment]
Sure, I can

**Comment by 1480c1** [comment]
Probably might need to ask someone with windows !10 to test it as well though

**Comment by 1480c1** [comment]
Well, the stuff was merged into msys2's stuff, but of course not upstream, going to close for now though

**Comment by Biswa96** [comment]
This patch had been applied in
* wine https://source.winehq.org/git/wine.git/commitdiff/2f05dd0719e7dc51b48397cf441e139a0c9c0ec3
* mingw-w64 https://sourceforge.net/p/mingw-w64/mingw-w64/ci/aaec44a4cea27f1ee1212d82c76e1ab510f7ed49/

**Comment by rmkimathi** [comment]
Thanks for all your efforts. Built m-ab-s mpv plays AV1 with hwdec dx11va like a champ on Windows 10  intel TGL Xe graphics. Very satisfying.
